dds/DCPS/DataReaderImpl.cpp

```diff
@@ -572,23 +572,19 @@ DataReaderImpl::remove_associations(const WriterIdSeq& writers,
 }
 
 void
-DataReaderImpl::remove_or_reschedule(const PublicationId& pub_id)
+DataReaderImpl::remove_publication(const PublicationId& pub_id)
 {
-  ACE_WRITE_GUARD(ACE_RW_Thread_Mutex, write_guard, this->writers_lock_);
-  WriterMapType::iterator where = writers_.find(pub_id);
-  if (writers_.end() != where) {
-    WriterInfo& info = *where->second;
-    WriterIdSeq writers;
-    push_back(writers, pub_id);
-    bool notify = info.notify_lost_;
-    if (info.removal_deadline_ < ACE_OS::gettimeofday()) {
-      write_guard.release();
-      remove_associations_i(writers, notify);
-    } else {
-      write_guard.release();
-      remove_associations(writers, notify);
+  WriterIdSeq writers;
+  bool notify = false;
+  {
+    ACE_WRITE_GUARD(ACE_RW_Thread_Mutex, write_guard, this->writers_lock_);
+    WriterMapType::iterator where = writers_.find(pub_id);
+    if (writers_.end() != where) {
```

Comment:
When the find() is not successful, writers.length() will be zero. The subsequent remove_associations_i() call would check if writers.length() is zero before it can continue processing. Therefore, the semantic would be equivalent if you consider both remove_publication() and remove_associations_i() together.